File: 0754-reach-a-number/0754-reach-a-number.cpp
class Solution {
public:
    int reachNumber(int target) {
        target = abs(target);
        int k = sqrt(target*2);
        int sum = getSum(k);
        while(sum < target) sum += (++k);
        const int d = sum - target;
        if(d % 2 == 0) return k;
        return (k+1) + (k%2);
    }
    
    int getSum(int k) {
        return (k+1)*k/2;
    }
};

File: 0754-reach-a-number/0754-reach-a-number.js
/**
 * @param {number} target
 * @return {number}
 */
var reachNumber = function(target) {
    target = Math.abs(target);
    let k = Math.floor(Math.sqrt(target*2));
    let sum = getSum(k);
    while(sum < target) {
        sum += (++k);
    }
    let d = sum - target;
    if(d % 2 == 0) return k;
    return k + 1 + (k%2);
};

var getSum = function(k) {
    return (k+1)*k/2;
}